lib/govuk_index/page_traffic_worker.rb

```diff
@@ -4,7 +4,14 @@ class PageTrafficWorker < Indexer::BaseWorker
     QUEUE_NAME = 'bulk'.freeze
     sidekiq_options queue: QUEUE_NAME
 
-    def perform(records, destination_index)
+    def self.perform_async(records, destination_index)
+      data = Base64.encode64(Zlib::Deflate.deflate(Sidekiq.dump_json(records)))
+      super(data, destination_index)
+    end
+
+    def perform(data, destination_index)
```

Comment:
I don't really know enough about what future developers might expect from a sidekiq job, but is it too weird to have the different signatures for `perform` and `perform_async`?

Reply:
I did look at using a Sidekiq middleware instead, but it was very easy to understand that compress was happening, and as every middleware is used by every worker that made it all worse. Should also check if `perform_now` would just break.